src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -43,330 +43,386 @@
+		// Set Gimbal pitch to track target in the center of the view
+		const float gimbal_height = calculateGimbalHeight((kFollowAltitudeMode)_param_flw_tgt_alt_m.get(), target_position_filtered(2));
+		gimbal_pitch = pointGimbalAt(drone_to_target_vector.norm(), gimbal_height);
 
-		if (_emergency_ascent) {
+		// Emergency ascent when too close to the ground
+		in_emergency_ascent = PX4_ISFINITE(_dist_to_ground) && _dist_to_ground < MINIMUM_SAFETY_ALTITUDE;
+		if (in_emergency_ascent) {
 			_position_setpoint(0) = _position_setpoint(1) = NAN;
 			_position_setpoint(2) = _position(2);
 			_velocity_setpoint(0) = _velocity_setpoint(1) = 0.0f;
 			_velocity_setpoint(2) = -EMERGENCY_ASCENT_SPEED; // Slowly ascend
 		}
 
-		// Yaw setpoint: Face the target
-		const Vector2f target_to_drone_xy = Vector2f(_position.xy()) - Vector2f(
-				target_position_filtered.xy());
+	} else {
+		// Target estimate is invalid : Stay in current position
+		_position_setpoint.setAll(NAN);
+		_velocity_setpoint.setAll(0.0f);
+	}
 
-		if (target_to_drone_xy.longerThan(MINIMUM_DISTANCE_TO_TARGET_FOR_YAW_CONTROL)) {
-			_yaw_setpoint = atan2f(-target_to_drone_xy(1), -target_to_drone_xy(0));
-		}
+	// Follow Target Status message for debugging
+	follow_target_status.timestamp = hrt_absolute_time();
 
-		// Gimbal setpoint
-		float gimbal_height = 0;
+	// Target position velocity filter output
+	_target_position_velocity_filter.getState().copyTo(follow_target_status.pos_est_filtered);
+	_target_position_velocity_filter.getRate().copyTo(follow_target_status.vel_est_filtered);
 
-		switch (_param_nav_ft_gmb_m.get()) {
-		case FOLLOW_GIMBAL_MODE_2D:
-			gimbal_height = -_position(2);
-			break;
+	// Orbit angle calculated related variables
+	follow_target_status.tracked_target_course = _target_course_rad;
+	follow_target_status.follow_angle = _follow_angle_rad;
+	follow_target_status.raw_orbit_angle_setpoint = raw_orbit_angle_setpoint;
+	follow_target_status.orbit_angle_setpoint = _orbit_angle_setpoint_rad;
 
-		case FOLLOW_GIMBAL_MODE_3D:
-			// Point the gimbal at the target's 3D coordinates
-			gimbal_height = -(_position(2) - (target_position_filtered(2)));
-			break;
+	// [Debug] Log orbit angle max & actual rate setpoint
+	follow_target_status.max_angular_rate_setpoint = _orbit_angle_traj_generator.getVelSp();
+	follow_target_status.angular_rate_setpoint = _orbit_angle_traj_generator.getCurrentVelocity();
 
-		case FOLLOW_GIMBAL_MODE_2D_WITH_TERRAIN:
-			// Point the gimbal at the ground level in this tracking mode
-			gimbal_height = _dist_to_ground;
-			break;
-
-		default:
-			break;
-		}
+	// [Debug] Log desired Raw Follow position
+	const Vector3f drone_desired_position_raw = calculateDesiredDronePosition(_target_position_velocity_filter.getState(), _target_course_rad + _follow_angle_rad);
+	drone_desired_position_raw.copyTo(follow_target_status.desired_position_raw);
 
-		point_gimbal_at(target_to_drone_xy.norm(), gimbal_height);
+	// Etc
```

Comment:
Etc?

Reply:
Removed, it was meant to say : "Other things to log .."